### PR TITLE
update pytests action for main branch

### PR DESCRIPTION
**Pull request by ianhi**

### CODE REVIEW

**Comment by ianhi** [comment]
That looks like a real error in the test

**Comment by ianhi** [comment]
or rather in the package - exposed by the test

**Comment by ianhi** [comment]
@henrypinkard latest commit should be a fix. I think you just released a version with a broken requirement that will probably fail on import for most users. may be good to release a patch release quick after merging this

File: setup.py [hunk 1]
```diff
@@ -21,7 +21,12 @@
     long_description_content_type="text/markdown",
     url="https://github.com/micro-manager/pycro-manager",
     packages=setuptools.find_packages(),
-    install_requires=["numpy", "dask[array]>=2.4.0", "zmq"],
+    install_requires=[
+        "numpy",
+        "dask[array]>=2.4.0",
```
Comment (ianhi) [inline setup.py:26]: this is different from requirements.txt  - does it need to be updated as well?

**Comment by henrypinkard** [comment]
Good catch! I'm still a bit confused about the requirements.txt and setup.py difference, as well as `dask[array]` which I think maybe is gone now in favor of just `dask`, as well as why `dask 2022.2.1` exists on pypi but seems to break the tests

**Comment by ianhi** [comment]
So the most up to date thing to do is to:

move everything in setup.py into setup.cfg

specify the build requirements in pyproject.toml
eliminate setup.py

remove requirements.txt and instead use the concept of `extra_requires`

**Comment by ianhi** [comment]
like so: https://github.com/tlambert03/pymmcore-plus/blob/62416c797ec40a7c7614fc16061f1928ced8382d/setup.cfg#L44-L61

**Comment by ianhi** [comment]
build system stuff, you can get rid fo setuptoolsscm bc you don;t use that.

https://github.com/tlambert03/pymmcore-plus/blob/62416c797ec40a7c7614fc16061f1928ced8382d/pyproject.toml#L1-L3

**Pull request merged**